tsdb/store.go

```diff
@@ -363,136 +375,159 @@ func (s *Store) loadShards() error {
 	log, logEnd := logger.NewOperation(s.Logger, "Open store", "tsdb_open")
 	defer logEnd()
 
+	shardLoaderWg := new(sync.WaitGroup)
 	t := limiter.NewFixed(runtime.GOMAXPROCS(0))
 	resC := make(chan *res)
```

Comment:
I know this was already in the code, but not having a buffer size on `resC` might not be ideal now that we are outputting something in the log after opening each shard. `CompletedShard` is called after sending a `res` to `resC`. This means after the first shard is completed, no more calls to `CompletedShard` will be made until `walkShardsAndProcess` completes. If you a few large shards, you probably won't notice. However, if you have a large number of small shards, you would see a period of no activity, followed by suddenly all the shards completing at once, even though that's not actually what happened. 

One solution is to call `CompletedShard` before sending to `resC`, but I think creating `resC` with a buffer size equal to number of shards is a superior solution. Sure, the channel buffer will consume more memory, but this will also allow the goroutines spawned when opening the shards to exit as soon as they are completed instead of blocking on sending to `resC`. This should reduce the overall memory required.